reports/hosted-services/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"log"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/ministryofjustice/cloud-platform-environments/pkg/authenticate"
+	"github.com/ministryofjustice/cloud-platform-how-out-of-date-are-we/reports/pkg/hoodaw"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/networking/v1beta1"
+	"k8s.io/client-go/kubernetes"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// resourceMap is used to store both string:string and string:map[string]string key
+// value pairs. The HOODAW API requires the first entry of map to contain a string:string,
+// the rest of the map consists of a primary key (string) with a value containing a map (string:string)
+// of Kubernetes namespaces and hostnames that don't contain the annotation contained within
+// the variable 'annotation'.
+type resourceMap map[string]interface{}
+
+// Namespace describes a Cloud Platform namespace object.
+type namespace struct {
+	Application      string
+	BusinessUnit     string
+	DeploymentType   string
+	Cluster          string
+	DomainNames      []string
+	GithubURL        string
+	Name             string
+	RbacTeam         []string
+	TeamName         string
+	TeamSlackChannel string
+}
```

Comment:
With the DRY principle in mind. Is it worth using the data type already declared here?
https://github.com/ministryofjustice/cloud-platform-environments/blob/main/pkg/namespace/namespace.go#L21-L32

Reply:
Updated the code